src/main/java/org/dependencytrack/resources/v1/ComponentResource.java

```diff
@@ -316,7 +321,15 @@ public Response createComponent(@PathParam("uuid") String uuid, Component jsonCo
             component.setNotes(StringUtils.trimToNull(jsonComponent.getNotes()));
 
             component = qm.createComponent(component, true);
-            kafkaEventDispatcher.dispatchBlocking(new ComponentRepositoryMetaAnalysisEvent(component));
+            ComponentProjection componentProjection =
```

Comment:
purl coordinates will not have the qualifier information which will be required for integrity data. We would need the canonicalized form of purl i think.

Comment in Component model:

 `private String purlCoordinates; // Field should contain only type, namespace, name, and version. Everything up to the qualifiers`